moncli/types.py

```diff
@@ -211,4 +214,78 @@ def validate_timezone(self, value):
         try:
             pytz.timezone(value)
         except (UnknownTimeZoneError):
-            raise ValidationError('Unknown time zone "{}".'.format(value))
\ No newline at end of file
+            raise ValidationError('Unknown time zone "{}".'.format(value))
+
+class StatusType(MondayType):
+    native_type = str
+    allow_casts = (int, str)
+    null_value = {}
+    def __init__(self, id: str = None, title: str = None,data_mapping:dict =None ,  *args, **kwargs):
+        if data_mapping:
+            values= [value for value in data_mapping.values()]        
+            self.native_type = values[0].__class__
+            self.choices = values
+        self._data_mapping =  data_mapping 
+        super().__init__(id=id, title=title, *args, **kwargs)
+    
+    def _process(self, value):
+        labels = self.metadata['labels']
+        if self.native_type == str:
+            try:
+                if isinstance(int(value),int):
+                    if value in labels.values():
+                        return value
+            except ValueError:
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+            if isinstance(value,int):
+                if value in labels.values():
+                    return value
+        return value
+
+    def _cast(self, value):
+        labels = self.metadata['labels']
+        label = str(value)
+        if self.native_type == str and isinstance(value,int):
+            if not (label in labels.keys()):
+                raise ConversionError('Cannot find status label with index "{}".'.format(value))
+            return labels[label]
+            
+        if isinstance(self.native_type,EnumMeta) and isinstance(value,str):
+            try:
+                value = int(value)
+                if not (label in labels.keys()):
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+                return labels[label]
+            except ValueError:
+                if not (label in labels.values()):
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+                return self._data_mapping[label]
+        if isinstance(self.native_type,EnumMeta) and isinstance(value,int):
+            if not (label in labels.keys()):
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+            return self._data_mapping[labels[label]]
+
+    def _export(self, value):
+        labels = self.metadata['labels']
+        label = str(value)
+        index  = None
+        if isinstance(self.native_type,str):
+            index = labels[label]
```

Comment:
This also does not look to be the correct way to get the index value when a str label is passed in.  You will need to do what you are doing at line 279 below for this.